eval/topdown.go

```diff
@@ -193,16 +224,14 @@ func evalContext(ctx *TopDownContext, iter TopDownIterator) error {
 
 func evalContextNegated(ctx *TopDownContext, iter TopDownIterator) error {
 
-	negation := &TopDownContext{
-		Query:    opalog.Body([]*opalog.Expr{ctx.Current().Complement()}),
-		Bindings: ctx.Bindings,
-		Previous: ctx,
-		Store:    ctx.Store,
-	}
+	negation := *ctx
```

Comment:
This comment belongs in the other routines...Is this creating a copy of ctx?  Maybe we should add some tests to ensure we're copying ctx wherever we need to.